### DISCUSSION TITLE
Tips for large schemas? Maybe some guides?

### DISCUSSION THREAD

**Discussion by rsinohara** [colinhacks/zod#711] [Q&A]
Hello,

Is there any guide on how to organize large schemas?

I am really unsure how to
- Compose and reuse schemas that are related
- Avoid repetition, without creating too much of a 'wildcard'
- Make sure, when I compose or reuse schemas, that I don't create combinations that would actually be wrong.

Basically, I am finding it harder to organize my schemas than I do with typescript types. Maybe because errors in typescript, usually, break things immediately, while with Zod I only notice at runtime.

Any tips are welcome.

**Comment by JacobWeisenburger** [comment]
A lot of those things are figured out by reading the docs and then playing around with zod to see how things work. I'm not sure that there are documented best practices for how to organize schemas.

If you provide a more specific example of what you are trying to do, it's more likely that someone will be able to answer your question.

Hope this helps. 😎
